zerver/views/streams.py

```diff
@@ -533,6 +533,7 @@ def you_were_just_subscribed_message(
 EMPTY_PRINCIPALS: Union[Sequence[str], Sequence[int]] = []
 
 
+@transaction.atomic
```

Comment:
Let's use `(durable=True)` here; we generally want to use that or `savepoint=False`, depending on whether we expect a particular transaction to be wrapped in an outer transaction. (Under no circumstances do we want to allow savepoints to be used, and both of these operators help achieve that)

Reply:
We need `savepoint = False` as it is wrapped by another transaction in `update_subscriptions_backend`